From 20fe58ba4ad024cc85fb9642572d82d846c239f4 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Mon, 16 Jun 2008 10:33:41 +0800
Subject: [PATCH] ENGR00080398 V4L output input crop issue

The input width after crop should be aligned with
16 or 8 pixels, at the same time, the burst size of
IC IDMA should equal 16 or 8.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/mxc/ipu3/ipu_common.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_common.c b/drivers/mxc/ipu3/ipu_common.c
index 85de798..3215c65 100644
--- a/drivers/mxc/ipu3/ipu_common.c
+++ b/drivers/mxc/ipu3/ipu_common.c
@@ -634,7 +634,10 @@ int32_t ipu_init_channel_buffer(ipu_channel_t channel, ipu_buffer_t type,
 	/* IC and ROT channels have restriction of 8 or 16 pix burst length */
 	if ((channel == MEM_PRP_ENC_MEM) || (channel == MEM_PRP_VF_MEM) ||
 	    (channel == MEM_PP_MEM)) {
-		_ipu_ch_param_set_burst_size(dma_chan, 16);
+		if ((width % 16) == 0)
+			_ipu_ch_param_set_burst_size(dma_chan, 16);
+		else
+			_ipu_ch_param_set_burst_size(dma_chan, 8);
 	} else if ((channel == MEM_ROT_ENC_MEM) ||
 		   (channel == MEM_ROT_VF_MEM) || (channel == MEM_ROT_PP_MEM)) {
 		_ipu_ch_param_set_burst_size(dma_chan, 8);
-- 
1.5.4.4

